contracts/SBTv3.scilla

```diff
@@ -396,12 +401,39 @@ transition ChangeData(to: ByStr20 , data_uri: String)
   _tag: "ZSBT_ChangeDataeCallback";
   _recipient: _sender;
   _amount: Uint128 0;
-  token_owner: to
+  value: to
+  };
+  msgs = one_msg msg_to_sender;
+  send msgs
+end
+
+(* @param: to  - The address of the profile  *)
+(* @param: value  - Bool for achievement   *)
+(* @Requirements: *)
+(* - The contract must not be paused. Otherwise, it must throw `PausedError` *)
+(* - The contract must be called by the address that is passed in as Parameter *)
+transition Earn_achievement(to: ByStr20 , value: Uint256)
+  RequireNotPaused;
+  (*Check that address is valid*)
+  RequireValidDestination to;
+
+  RequireNotSelf to _sender;
+  (*Check that address has sbt*)
+  HasToken to;
+
+  achievements_bool[to] := value;
+  msg_to_sender = {
+  _tag: "ZSBT_EarnAchievementCallback";
+  _recipient: _sender;
+  _amount: Uint128 0;
+  user: to;
+  achievement_earned: value
   };
   msgs = one_msg msg_to_sender;
```

Comment:
Can emit event on successful obtaining of achievement for frontend to listen to?